packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -350,7 +350,11 @@ export function TransferPanelMain({
   // whenever the user changes the `amount` input, it should update the amount in browser query params as well
   useEffect(() => {
     setQueryParams({ amount })
-  }, [amount])
+
+    if (amount.toLowerCase() === 'max') {
```

Comment:
Optional - Can we use `"max"` as a type or enum here and reference that, instead of hardcoding the string 'max' everywhere?
The reason being we want to be able to infer from the type of query param itself that it only allows `number` or `MAX/max` as a valid value.